### DISCUSSION TITLE
VERY intensive cpu usage

### DISCUSSION THREAD

**Discussion by JustSimplyKyle** [JakeStanger/ironbar#176] [Q&A]
![image](https://github.com/JakeStanger/ironbar/assets/68589851/543240f2-5b09-4433-8c7e-a09473e0c7c0)
![image](https://github.com/JakeStanger/ironbar/assets/68589851/b540d97c-6aca-4f6c-a2bb-442a5c80dc89)
As always, [dotfiles](https://gitea.com/SimplyKyle/dotfiles/src/branch/main/home/.config/ironbar) are available

**Comment by JustSimplyKyle** [comment]
Using the default config has no such issue tho

**Reply by JustSimplyKyle** [reply]
But, the very "similar"(the logic is the same) config in eww has quite sane cpu usage

**Comment by JakeStanger** [comment]
You are running several scripts with a 0ms poll time. This means as soon as the script exits, it's spawned again, which involves creating a new `sh` process and is relatively slow/expensive.

I suspect if you bump those up a bit, you'll see CPU use go down. A value of 50ms-100ms will be quick enough to be barely noticeable and should reduce overhead.
